sanntis/modules/BGCdetection.py

```diff
@@ -93,57 +94,45 @@ def transformEmeraldHmm(self, hmmFile):
                 spl = l.split()
                 self.entriesDct.setdefault(spl[3], []).append(spl[0])
 
-    def transformCDSpredToCDScontigs(self, cdsPredFile, f):
+    def transformCDSpredToCDScontigs(self, cdsPredFile, file_format):
 
         if not os.path.isfile(cdsPredFile):
             log.exception(f"{cdsPredFile} file not found")
         
-        prodigal_formating_exception = False
+        recs = list(SeqIO.parse(open(cdsPredFile, "r"), file_format))
```

Comment:
You can move this to the for loop, so you don't have to load the whole sequence in memory (not a massive problem unless you bump into a humounguos file)